cmd/notation/internal/display/metadata/json/inspect.go

```diff
@@ -0,0 +1,204 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json
+
+import (
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/hex"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/notaryproject/notation-core-go/signature"
+	"github.com/notaryproject/notation-go/plugin/proto"
+	"github.com/notaryproject/notation/cmd/notation/internal/output"
+	"github.com/notaryproject/notation/internal/envelope"
+	"github.com/notaryproject/tspclient-go"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type inspectOutput struct {
+	MediaType  string `json:"mediaType"`
+	Signatures []Signature
+}
+
+// Signature is the signature envelope for printing in human readable format.
+type Signature struct {
+	MediaType             string             `json:"mediaType"`
+	Digest                string             `json:"digest,omitempty"`
+	SignatureAlgorithm    string             `json:"signatureAlgorithm"`
+	SignedAttributes      map[string]any     `json:"signedAttributes"`
+	UserDefinedAttributes map[string]string  `json:"userDefinedAttributes"`
+	UnsignedAttributes    map[string]any     `json:"unsignedAttributes"`
+	Certificates          []Certificate      `json:"certificates"`
+	SignedArtifact        ocispec.Descriptor `json:"signedArtifact"`
+}
+
+// Certificate is the certificate information for printing in human readable
+// format.
+type Certificate struct {
+	SHA256Fingerprint string    `json:"SHA256Fingerprint"`
+	IssuedTo          string    `json:"issuedTo"`
+	IssuedBy          string    `json:"issuedBy"`
+	Expiry            time.Time `json:"expiry"`
+}
+
+// Timestamp is the timestamp information for printing in human readable.
+type Timestamp struct {
+	Timestamp    string        `json:"timestamp,omitempty"`
+	Certificates []Certificate `json:"certificates,omitempty"`
+	Error        string        `json:"error,omitempty"`
+}
+
+// InspectHandler is the handler for inspecting metadata information and
+// rendering it in JSON format.
+type InspectHandler struct {
+	printer *output.Printer
+
+	output inspectOutput
+}
+
+// NewInspectHandler creates a new InspectHandler.
+func NewInspectHandler(printer *output.Printer) *InspectHandler {
+	return &InspectHandler{
+		printer: printer,
+	}
+}
+
+// OnReferenceResolved sets the artifact reference and media type for the
+// handler.
+//
+// the reference is no-op for this handler.
+func (h *InspectHandler) OnReferenceResolved(reference, mediaType string) {
```

Comment:
You may use `_` to indicate it is a no-op.
```suggestion
func (h *InspectHandler) OnReferenceResolved(_, mediaType string) {
```